Automattic/wp-calypso
# Shopping Cart: Clean up lib/cart-values and remove unused functions, take 2

#### Changes proposed in this Pull Request

This is the second attempt at https://github.com/Automattic/wp-calypso/pull/52211 which was reverted in https://github.com/Automattic/wp-calypso/pull/52351 due to problems with the signup cart.

In https://github.com/Automattic/wp-calypso/pull/51105 we removed a bunch of unused code from lib/cart-values. This PR does a little refactoring to remove some more. It does the following:

- Remove `addCartItem`. It was only used in signup and adding products to a cart manually is trivial.
- Remove `getAllCartItemsSorted`. It was only used for the popover cart and now that the shopping-cart endpoint does sorting it's not really needed.
- Remove `sortProducts` (from calypso-products). This function was originally for the cart, but now it was only used by the purchases selector. This seems unnecessary.
- Remove `preprocessCartForServer`. It was only used in signup and didn't actually do anything but pull in the cart and spit it out.
- Remove `fillInAllCartItemAttributes`. It was only used in signup and is not necessary when the rest of calypso uses `fillInSingleCartItemAttributes`.
- Remove the call to `extendImmutabilityHelper` in cart-values. Side effects are dangerous and it's not even clear what that was doing. It was probably something to do with the calls to `update` in that file, but since they have all been removed, that side-effect was also removed.
- Remove types from lib/cart-values. They were already deprecated in favor of the types in `@automattic/shopping-cart`.
- Remove lodash from lib/cart-values. All the uses were easily handled by native JS.
- Add additional tests for lib/signup/cart and lib/cart-values (to cover the issues encountered in the first version of this PR).

#### Testing instructions

- Verify that the removed functions mentioned above are not used anywhere.
- Visit `/start` and create a new site with any paid plan. Verify that the flow works correctly and that the site is created.
